support/crown/src/common.rs

```diff
@@ -101,12 +101,11 @@ fn find_primitive_impls<'tcx>(tcx: TyCtxt<'tcx>, name: &str) -> impl Iterator<It
         _ => {
             return Result::<_, rustc_errors::ErrorGuaranteed>::Ok(&[] as &[_])
                 .into_iter()
-                .flatten()
                 .copied();
         },
     };
 
-    tcx.incoherent_impls(ty).into_iter().flatten().copied()
+    tcx.incoherent_impls(ty).into_iter().copied()
```

Comment:
Are you able to build crown with just these two changes? On NixOS, I am seeing similar compilation errors related to other usages of `flatten` in this file. Also, there are errors about bounds not being met for the return types in `find_primitive_impls`. 

```
review on  rustc-1.83 [$!?] is 📦 v0.0.1 via 🐍 v3.11.10 via 🦀 v1.83.0 took 21s 
❯ ./mach build      
NOTE: Entering nix-shell ./shell.nix
this derivation will be built:
  /nix/store/7c2hd97q811dxb68ly2nj7avixgvd52l-crown.drv
building '/nix/store/7c2hd97q811dxb68ly2nj7avixgvd52l-crown.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/33sjmmb7bp7f6drs3v4cplic2n231xxw-crown
source root is crown
Executing cargoSetupPostUnpackHook
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
Executing cargoSetupPostPatchHook
Validating consistency between /build/crown/Cargo.lock and /build/cargo-vendor-dir/Cargo.lock
Finished cargoSetupPostPatchHook
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
Running phase: buildPhase
Executing cargoBuildHook
cargoBuildHook flags: -j 24 --target x86_64-unknown-linux-gnu --offline --profile release
   Compiling crown v0.0.1 (/build/crown)
error[E0277]: the trait bound `[_]: std::marker::Copy` is not satisfied
   --> src/common.rs:104:18
    |
104 |                 .copied();
    |                  ^^^^^^ the trait `std::marker::Copy` is not implemented for `[_]`
    |
    = help: the trait `std::marker::Copy` is implemented for `[T; N]`
note: required by a bound in `copied`
   --> /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/iter/traits/iterator.rs:3417:5

error[E0277]: the trait bound `[_]: std::marker::Copy` is not satisfied
  --> src/common.rs:74:65
   |
74 | fn find_primitive_impls<'tcx>(tcx: TyCtxt<'tcx>, name: &str) -> impl Iterator<Item = DefId> + 'tcx {
   |                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::marker::Copy` is not implemented for `[_]`, which is required by `Copied<std::result::IntoIter<&[_]>>: std::iter::Iterator`
   |
   = help: the trait `std::marker::Copy` is implemented for `[T; N]`
   = note: required for `Copied<std::result::IntoIter<&[_]>>` to implement `std::iter::Iterator`

error[E0271]: expected `IntoIter<&[_]>` to be an iterator that yields `&DefId`, but it yields `&[_]`
  --> src/common.rs:74:65
   |
74 | fn find_primitive_impls<'tcx>(tcx: TyCtxt<'tcx>, name: &str) -> impl Iterator<Item = DefId> + 'tcx {
   |                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&[_]`, found `&DefId`
   |
   = note: expected reference `&[_]`
              found reference `&rustc_span::def_id::DefId`
   = note: required for `Copied<std::result::IntoIter<&[_]>>` to implement `std::iter::Iterator`

error[E0308]: mismatched types
   --> src/common.rs:108:5
    |
108 |     tcx.incoherent_impls(ty).into_iter().copied()
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Copied<IntoIter<&[_]>>`, found `Copied<Iter<'_, DefId>>`
    |
    = note: expected struct `Copied<std::result::IntoIter<&[_]>>`
               found struct `Copied<std::slice::Iter<'_, rustc_span::def_id::DefId>>`

error[E0277]: `&rustc_span::def_id::DefId` is not an iterator
   --> src/common.rs:237:22
    |
237 |                     .flatten()
    |                      ^^^^^^^ `&rustc_span::def_id::DefId` is not an iterator
    |
    = help: the trait `std::iter::Iterator` is not implemented for `&rustc_span::def_id::DefId`, which is required by `&rustc_span::def_id::DefId: IntoIterator`
    = note: required for `&rustc_span::def_id::DefId` to implement `IntoIterator`
note: required by a bound in `flatten`
   --> /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/iter/traits/iterator.rs:1539:5

error[E0599]: the method `flat_map` exists for struct `Flatten<Iter<'_, DefId>>`, but its trait bounds were not satisfied
   --> src/common.rs:238:22
    |
234 |                   let inherent_impl_children = tcx
    |  ______________________________________________-
235 | |                     .inherent_impls(def_id)
236 | |                     .into_iter()
237 | |                     .flatten()
238 | |                     .flat_map(|&impl_def_id| item_children_by_name(tcx, impl_def_id, segment));
    | |                     -^^^^^^^^ method cannot be called on `Flatten<Iter<'_, DefId>>` due to unsatisfied trait bounds
    | |_____________________|
    |
    |
    = note: the following trait bounds were not satisfied:
            `<&rustc_span::def_id::DefId as IntoIterator>::IntoIter = _`
            which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
            `<&rustc_span::def_id::DefId as IntoIterator>::Item = _`
            which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
            `&rustc_span::def_id::DefId: IntoIterator`
            which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
            `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
            which is required by `&mut Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`

error[E0277]: `&rustc_span::def_id::DefId` is not an iterator
   --> src/common.rs:282:18
    |
282 |                 .flatten()
    |                  ^^^^^^^ `&rustc_span::def_id::DefId` is not an iterator
    |
    = help: the trait `std::iter::Iterator` is not implemented for `&rustc_span::def_id::DefId`, which is required by `&rustc_span::def_id::DefId: IntoIterator`
    = note: required for `&rustc_span::def_id::DefId` to implement `IntoIterator`
note: required by a bound in `flatten`
   --> /rustc/90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf/library/core/src/iter/traits/iterator.rs:1539:5

error[E0599]: the method `flat_map` exists for struct `Flatten<Iter<'_, DefId>>`, but its trait bounds were not satisfied
   --> src/common.rs:283:18
    |
279 |               let inherent_impl_children = tcx
    |  __________________________________________-
280 | |                 .inherent_impls(def_id)
281 | |                 .into_iter()
282 | |                 .flatten()
283 | |                 .flat_map(|&impl_def_id| item_children_by_name(tcx, impl_def_id, segment));
    | |                 -^^^^^^^^ method cannot be called on `Flatten<Iter<'_, DefId>>` due to unsatisfied trait bounds
    | |_________________|
    |
    |
    = note: the following trait bounds were not satisfied:
            `<&rustc_span::def_id::DefId as IntoIterator>::IntoIter = _`
            which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
            `<&rustc_span::def_id::DefId as IntoIterator>::Item = _`
            which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
            `&rustc_span::def_id::DefId: IntoIterator`
            which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
            `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
            which is required by `&mut Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`

Some errors have detailed explanations: E0271, E0277, E0308, E0599.
For more information about an error, try `rustc --explain E0271`.
error: could not compile `crown` (bin "crown") due to 8 previous errors
error: builder for '/nix/store/7c2hd97q811dxb68ly2nj7avixgvd52l-crown.drv' failed with exit code 101;
       last 25 log lines:
       >    --> src/common.rs:283:18
       >     |
       > 279 |               let inherent_impl_children = tcx
       >     |  __________________________________________-
       > 280 | |                 .inherent_impls(def_id)
       > 281 | |                 .into_iter()
       > 282 | |                 .flatten()
       > 283 | |                 .flat_map(|&impl_def_id| item_children_by_name(tcx, impl_def_id, segment));
       >     | |                 -^^^^^^^^ method cannot be called on `Flatten<Iter<'_, DefId>>` due to unsatisfied trait bounds
       >     | |_________________|
       >     |
       >     |
       >     = note: the following trait bounds were not satisfied:
       >             `<&rustc_span::def_id::DefId as IntoIterator>::IntoIter = _`
       >             which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
       >             `<&rustc_span::def_id::DefId as IntoIterator>::Item = _`
       >             which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
       >             `&rustc_span::def_id::DefId: IntoIterator`
       >             which is required by `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
       >             `Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
       >             which is required by `&mut Flatten<std::slice::Iter<'_, rustc_span::def_id::DefId>>: std::iter::Iterator`
       >
       > Some errors have detailed explanations: E0271, E0277, E0308, E0599.
       > For more information about an error, try `rustc --explain E0271`.
       > error: could not compile `crown` (bin "crown") due to 8 previous errors
       For full logs, run 'nix log /nix/store/7c2hd97q811dxb68ly2nj7avixgvd52l-crown.drv'.

```

Copying the same code from clippy does fix these bound error. Here is the diff that works for me:

```
diff --git a/support/crown/src/common.rs b/support/crown/src/common.rs
index 5aad3075c5..e36343061b 100644
--- a/support/crown/src/common.rs
+++ b/support/crown/src/common.rs
@@ -99,13 +99,11 @@ fn find_primitive_impls<'tcx>(tcx: TyCtxt<'tcx>, name: &str) -> impl Iterator<It
         "f64" => SimplifiedType::Float(FloatTy::F64),
         #[allow(trivial_casts)]
         _ => {
-            return Result::<_, rustc_errors::ErrorGuaranteed>::Ok(&[] as &[_])
-                .into_iter()
-                .copied();
+            return [].iter().copied();
         },
     };
 
-    tcx.incoherent_impls(ty).into_iter().copied()
+    tcx.incoherent_impls(ty).iter().copied()
 }
 
 fn non_local_item_children_by_name(tcx: TyCtxt<'_>, def_id: DefId, name: Symbol) -> Vec<Res> {
@@ -234,7 +232,6 @@ pub fn def_path_res(cx: &LateContext<'_>, path: &[&str]) -> Vec<Res> {
                 let inherent_impl_children = tcx
                     .inherent_impls(def_id)
                     .into_iter()
-                    .flatten()
                     .flat_map(|&impl_def_id| item_children_by_name(tcx, impl_def_id, segment));
 
                 let direct_children = item_children_by_name(tcx, def_id, segment);
@@ -279,7 +276,6 @@ pub fn def_local_res(cx: &LateContext<'_>, path: &str) -> Vec<Res> {
             let inherent_impl_children = tcx
                 .inherent_impls(def_id)
                 .into_iter()
-                .flatten()
                 .flat_map(|&impl_def_id| item_children_by_name(tcx, impl_def_id, segment));
 
             let direct_children = item_children_by_name(tcx, def_id, segment);
```